src/natcap/invest/validation.py

```diff
@@ -193,9 +193,26 @@ def check_directory(dirpath, must_exist=True, permissions='rx', **kwargs):
                 dirpath = parent
                 break
 
-    permissions_warning = check_permissions(dirpath, permissions, True)
-    if permissions_warning:
-        return permissions_warning
+    MESSAGE_KEY = 'NEED_PERMISSION_DIRECTORY'
+
+    if 'r' in permissions:
+        try:
+            os.scandir(dirpath).close()
+        except:
+            return MESSAGES[MESSAGE_KEY].format(permission='read')
+
+    # Check for x access before checking for w, since w operations to a dir are dependent on x access
+    if 'x' in permissions:
+        try:
+            os.chdir(dirpath)
+        except:
+            return MESSAGES[MESSAGE_KEY].format(permission='execute')
```

Comment:
This seems to be the problem. If I comment out this block, all the other test suites pass. A simple change of directory seemed innocuous at the time, but obviously it is creating side effects!